pip/commands/install.py

```diff
@@ -251,12 +251,12 @@ def run(self, options, args):
                     progress_bar=options.progress_bar,
                 )
 
-                self.populate_requirement_set(
-                    requirement_set, args, options, finder, session, self.name,
-                    wheel_cache
-                )
-
                 try:
+                    self.populate_requirement_set(
```

Comment:
`AssertionError: temp files not expected`

I think it's like this:

* Before the code in question, a temporary directory has been created to store ephemeral wheels. This is new in this PR.
* The tests you see failing are those which exercise an error case, e.g. no requirements specified.
* If `self.populate_requirement_set(...)` fails, we want to call the cleanup code to remove that temporary dir.
* The cleanup code is in the `finally` block, so we need the step that can fail to be in the `try`.

The change in this PR is that a temporary directory is created before this; previously it only created temporary files later, so there was no need for cleanup if the earlier stage failed.